### ISSUE TITLE
NetworkConnection.Objects is empty 

### ISSUE THREAD

**Issue by TakeruGrima** [FirstGearGames/FishNet#176]
1. Spawn a player prefab when a client connect to a server
2. When the client disconnect remove the client ownership
3. The player prefab would stay in the scene and will be active
4. Connect the client again and don't spawn a player prefab
5. On the player connection give the player prefab ownership to the new client
6. Wait the client to finish all initialization
7. The Objects list on the NetworkConnection will be empty on the client but not on the server

It's an issue if we want to be able to get the NetworkObject of the player's prefab using the NetworkConnection. In our logic we make sure that the FirstObject is the player and we get the NetworkObject Id to check if a not networked script should do a specific task.

**Comment by maxkratt** [comment]
To add to this, here is a snippet of code from the PlayerSpawner component that I have edited. Whenever the new client is given the ownership of an existing object instead of spawning a new one, the NetworkConnection.Objects list on the client is not updated to include the object.
```cs
private void SceneManager_OnClientLoadedStartScenes(NetworkConnection conn, bool asServer)
{
    if (!asServer)
        return;
    if (_playerPrefab == null)
    {
        Debug.LogWarning($"Player prefab is empty and cannot be spawned for connection {conn.ClientId}.");
        return;
    }

    Vector3 position;
    Quaternion rotation;
    SetSpawn(_playerPrefab.transform, out position, out rotation);

    NetworkObject nob;
    GameObject obj = GameObject.Find("player");

    if (obj)
    {
        nob = obj.GetComponent<NetworkObject>();
        nob.GiveOwnership(conn);
    }
    else
    {
        nob = _networkManager.GetPooledInstantiated(_playerPrefab, true);
        nob.transform.SetPositionAndRotation(position, rotation);
        _networkManager.ServerManager.Spawn(nob, conn);
    }


    //If there are no global scenes 
    if (_addToDefaultScene)
        _networkManager.SceneManager.AddOwnerToDefaultScene(nob);

    OnSpawned?.Invoke(nob);
}
```
I have tested it in FN 1.6.4 and 1.5.10

**Comment by FirstGearGames** [comment]
1.6.4 and 1.5.10?

**Comment by maxkratt** [comment]
Whoops, I mean 2.6.4 and 2.5.10. Also removing the scene condition resolves the issue.

**Comment by TakeruGrima** [comment]
I tried to remove the scene condition and it doesn't resolves the issue for me with the version 2.6.4 Pro

**Comment by FirstGearGames** [comment]
The latest version of 2x is 2.6.6. Please see if the issue exist there.

**Comment by TakeruGrima** [comment]
Still have the same issue with 2.6.6

**Comment by TakeruGrima** [comment]
I found what is the issue on my side.
When I give the ownership  to a connection, in networkObject.GiveOwnership(NetworkConnection connection) on the client side the previous Owner & the new Owner are the same so in the part of the code bellow :
`            if (asServer || !NetworkManager.IsHost)
            {
                if (activeNewOwner)
                    newOwner.AddObject(this);
                if (prevOwner.IsValid)
                    prevOwner.RemoveObject(this);
            }`

The object is added to the newOwner connection but removed right after because the 2 connections are the same, so I edited this part like this :
`            if (asServer || !NetworkManager.IsHost)
            {
                if (activeNewOwner)
                    newOwner.AddObject(this);
                if (prevOwner.IsValid && prevOwner != newOwner)
                    prevOwner.RemoveObject(this);
            }`

**Comment by FirstGearGames** [comment]
Resolved in 3.2.1
